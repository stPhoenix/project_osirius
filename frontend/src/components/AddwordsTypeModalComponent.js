import React from 'react';
import {Modal, ModalBody, Card, CardBody, CardTitle} from 'reactstrap';


export const AddwordsTypeModalComponent = (props) => {
    return (
        <Modal isOpen={props.modal} toggle={props.toggle_modal} className="p-0 align-self-center">
            <ModalBody className="p-0">
                <Card className="rounded-0">
                    <CardBody className="d-flex flex-column">
                        <CardTitle>Choose category</CardTitle>
                        
                            <form>
                                <select className="custom-form-control" id="learn_language" onChange={props.handleChange} name="category">
                                        {props.cats.map((category) => (<option key={category.name}>{category.name}</option>))}
                                </select>
                            </form>    
                        <button className="btn btn-link align-self-end" onClick={props.add_custom}>Add</button>
                    </CardBody>
                </Card>
            </ModalBody>
        </Modal>
    );
};