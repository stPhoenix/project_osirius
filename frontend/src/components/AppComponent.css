.theme-primary{
    background-color: #009be5;
}
.theme-light{
    background-color: #63ccff;
}
.theme-dark{
    background-color: #006db3;
}