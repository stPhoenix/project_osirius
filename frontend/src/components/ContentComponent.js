import React from 'react';


export const ContentComponent = (props) => {
    return(
        <div className="col-10 d-flex flex-column align-self-center align-items-center justify-content-center p-0">
            {props.children}
        </div>
    );
};