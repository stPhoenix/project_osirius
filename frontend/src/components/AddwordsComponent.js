import React from 'react';
import {Link, Switch, Route} from 'react-router-dom';
import typing from '../assets/typing.png';
import presets from '../assets/presets.png';
import './AppComponent.css';
import AddwordsType from '../containers/AddwordsType';
import AddwordsPresets from '../containers/AddwordsPresets';
import {ContentComponent} from '../components';
import PropTypes from 'prop-types';

const componentProps = {
  location: PropTypes.object.isRequired,  
};

export const AddwordsComponent = ({location}) => {
	if (location.pathname !== "/addwords"){
		return (<Switch>
					<Route path="/addwords/type" component={AddwordsType} />
					<Route path="/addwords/presets" component={AddwordsPresets} />
				</Switch>);	
	};
	return(
    <ContentComponent>
            <section className="d-flex flex-column align-items-center justify-content-center">
		      <div className="d-flex flex-column align-items-center justify-content-center">
                <img src={typing} alt="typing" style={{height:'7rem', width:'auto'}} />
		        <h4 className="opacity-50 text-center px-2 mb-4 mt-0">Type word. System will search word in database or in the internet.</h4>
		        <Link className="custom-btn theme-primary px-3 py-1 mb-5 black" to="/addwords/type" > Add words</Link>
              </div>
              <div className="d-flex flex-column align-top align-items-center justify-content-center">
                <img src={presets} alt="presets" className="" style={{height:'6rem', width:'auto'}} />
		        <h4 className="opacity-50 text-center px-2 mb-4 mt-0">Choose word from preset in database.</h4>
		        <Link className="custom-btn theme-primary px-3 py-1 mb-3 black" to="/addwords/presets" > Add words</Link>
              </div>
            </section>
	</ContentComponent>
		);
};

AddwordsComponent.propTypes = componentProps;