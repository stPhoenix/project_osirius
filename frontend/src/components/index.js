import {AppComponent} from './AppComponent';
import {CustomAlertComponent} from './CustomAlertComponent';
import {LoginComponent} from './LoginComponent';
import {NavBarComponent} from './NavBarComponent';
import {NavItemComponent} from './NavItemComponent';
import {NewsComponent} from './NewsComponent';
import {SignUpComponent} from './SignUpComponent';

export {
    AppComponent,
    CustomAlertComponent,
    LoginComponent,
    NavBarComponent,
    NavItemComponent,
    NewsComponent,
    SignUpComponent
};