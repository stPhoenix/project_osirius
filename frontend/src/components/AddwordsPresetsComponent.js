import React from 'react';
import './AppComponent.css';
import {WordsTableComponent, ContentComponent} from '../components';


export const AddwordsPresetsComponent = (props) => {
    return (<ContentComponent>
                <form className="col-12 col-lg-8 px-0">
                    <label htmlFor="categories" >Choose category:</label>
                    <select className="custom-form-control" id="categories" onChange={props.handleChange} name="category">
                                        {props.cats.map((category) => (<option key={category.name}>{category.name}</option>))}
                    </select>
                </form>
                <div className="d-flex align-self-end mt-3 px-0" >
					<button className="btn btn-link " onClick={props.add_selected}>Add selected</button>
					<button className="btn btn-link pr-0" onClick={props.add_all}>Add all</button>
				</div>
                <WordsTableComponent words={props.words} handleCheck={props.handleCheck} />
          </ContentComponent>);  
};