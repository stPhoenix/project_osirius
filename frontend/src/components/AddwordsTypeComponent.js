import React from 'react';
import './AppComponent.css';
import {CardDeck, Card, CardBody, CardText} from 'reactstrap';
import {AddwordsTypeModalComponent} from '../components';


export const AddwordsTypeComponent = (props) => {
	
	let data = null;
	if (props.global_word_search === true){
		data = (props.words.map(word => {
			return (<Card key={word.id+"words"} className="shadow-sm rounded-0">
                        <CardBody className="d-flex flex-column">
                            <CardText className="text-center">
                                <br />{word.name}
                                <br />{word.pronunciation}
                                <br />Category: [{word.category.map(category => (category.name+" "))}]
                                <br />{word.translation}
                            </CardText>
                            <button name={word.id} className="btn btn-link align-self-end" onClick={props.add_global}>Add</button>
                        </CardBody>
                    </Card>);
		}));
	} else if(props.google_translate_search === true){
		data += (props.words.map(word => {
			return (<Card key={word.id+"words"} className="shadow-sm rounded-0">
                        <CardBody className="d-flex flex-column">
                            <CardText className="text-center">
                                <br />{word.name}
                                <br />{word.pronunciation}
                                <br />{word.translation}
                            </CardText>
                        <button className="btn btn-link align-self-end" onClick={(e) => {props.choose_category(word.name, word.pronuciation, word.translation)}}>Add</button>
                        </CardBody>
                    </Card>);
		}));
	}
	
	return(<div className="col-10 d-flex flex-column px-0 justify-content-center align-items-center">
				<form className="d-flex flex-column w-100">
					<div className="form-group">
						<label htmlFor="search">Type word:</label>
						<input id="search" type="text" className="custom-form-control" value={props.search_word} name="search_word" onChange={props.handleChange}/>
					</div>
                    <button className="custom-btn theme-primary mb-2 align-self-center" onClick={props.search}>Search</button>
				</form>
                <CardDeck className="w-100">
                    {data}
                </CardDeck>
                <AddwordsTypeModalComponent add_custom={props.add_custom}
                                            handleChange={props.handleChange}
                                            cats={props.cats}
                                            modal={props.modal}
                                            toggle_modal={props.toggle_modal} />
		</div>);
};