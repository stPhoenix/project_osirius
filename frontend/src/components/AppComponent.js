import React from 'react';
import {Container, Col, Row} from 'reactstrap';
import './AppComponent.css';
import logo from '../assets/logo.png';
import NavBar from '../containers/NavBar';
import Login from '../containers/Login';
import Alerts from '../containers/Alerts';
import {Switch, Route} from 'react-router';
import News from '../containers/News';
import Logout from '../containers/Logout';
import {SignUpComponent} from '../components';


export const AppComponent = () => {
        return (
      <Container fluid className="bg-first">
        <Row className="d-flex flex-column">
            <Col xs="12" className="px-0" style={{zIndex:1}}>
                <header className="theme-dark px-4 py-2">
                    <img style={{height:'2rem', width:'auto'}} src={logo} alt="logo" />
                </header>
            </Col>
            <Col xs="12" className="px-0" style={{zIndex:1}}>
                <section className="theme-primary shadow-sm full-screen">
                    <NavBar/>
                </section>
            </Col>
            <Col xs="12 px-0">
                <main className="d-flex flex-column full-screen">
                    <Alerts />
                    <section className="d-flex justify-content-center align-items-center">
                        <Switch>
                            <Route exact path="/" component={News} />
                            <Route path="/news" component={News} />
                            <Route path="/login" component={Login} />
                            <Route path="/logout" component={Logout} />
                            <Route path="/signup" component={SignUpComponent} />
                        </Switch>
                    </section>
                    
                </main>
            </Col>
        </Row>
        <Row className="d-flex flex-column">
            <Col xs="12">
                <footer>
                </footer>
            </Col>
        </Row>
      </Container>
    );
  };

