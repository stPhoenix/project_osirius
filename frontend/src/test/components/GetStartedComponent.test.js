import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {GetStartedComponent} from '../../components';


describe('GetStartedComponent', () => {
    it('should render correctly', () => {
        const output = shallow(<GetStartedComponent />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});