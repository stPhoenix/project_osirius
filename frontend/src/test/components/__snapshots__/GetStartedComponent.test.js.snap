// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStartedComponent should render correctly 1`] = `
<ContentComponent>
  <h2>
    Get started
  </h2>
  <CardColumns
    tag="div"
  >
    <Card
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        1. Sign Up
      </CardHeader>
      <CardBody
        tag="div"
      >
        <CardTitle
          tag="h5"
        >
          If you not registered yet click Sign up link at menu bar and register. Otherwise - go to step 2.
        </CardTitle>
        <CardText
          tag="p"
        >
          Sign up form is pretty simple. Aside from username, password and email fields, you should select your home language(the language you speak and understand) - foreign words translation will be on  this language, and choose learn language - language you want to learn.
        </CardText>
      </CardBody>
    </Card>
    <Card
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        2. Add words
      </CardHeader>
      <CardBody
        tag="div"
      >
        <CardTitle
          tag="h5"
        >
          So, after you sign up, or log in you can see additional links in menu bar. Click on add words link.
        </CardTitle>
        <CardText
          tag="p"
        >
          You can use two ways to add foreign words for learn - by typing(system will try find word in database or internet) or by choosing from presets. If you choose presets - page will show words by category. If no words for your language pair(home language - learn language) you won't see anything. As service was made by me, current presets have words for pair english - japanese. I highly advise contributing in github for filling presets by words on different languages. Othewise - add words by typing based on your own needs.
        </CardText>
      </CardBody>
    </Card>
    <Card
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        3. Learn words
      </CardHeader>
      <CardBody
        tag="div"
      >
        <CardTitle
          tag="h5"
        >
          Okay, you have added words and ready to insert them into your brain. In your menu bar you can see link learn words. Click on it.
        </CardTitle>
        <CardText
          tag="p"
        >
          Learn words page is represented like card with word on it. Aside from word, translation and pronucitaion fields it has two buttons: learned and not learned. If you think that you have remembered word - click learn and word will not appear more on this page. Otherwise - when you choose not learned - words will show as more times as you clicking not learned.
        </CardText>
      </CardBody>
    </Card>
    <Card
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        4. Train words
      </CardHeader>
      <CardBody
        tag="div"
      >
        <CardTitle
          tag="h5"
        >
          Now, when you have brief remembered words you should train them for not giving them run from your memory. At menu bar click on words trainer link.
        </CardTitle>
        <CardText
          tag="p"
        >
          Each exercise on words trainer page enough described, so I won't stop here explaining it to you. What should mention - when you training words on each play once you gave right answer - word won't display on this play.
        </CardText>
      </CardBody>
    </Card>
    <Card
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        5. Repeat step 2
      </CardHeader>
      <CardBody
        tag="div"
      >
        <CardTitle
          tag="h5"
        >
          Good! You have come to this step! I can consume that you undesrtood main logic of service. So what are you waiting for - repeat from step 2 and easy learn more foreign words!
        </CardTitle>
      </CardBody>
    </Card>
    <Card
      tag="div"
    >
      <CardHeader
        tag="div"
      >
        Additional advice 1
      </CardHeader>
      <CardBody
        tag="div"
      >
        <CardTitle
          tag="h5"
        >
          In settings you can add and remove learnt languages.
        </CardTitle>
        <CardText
          tag="p"
        >
           Way of doing it pretty intuitive. Remember, all words in pages Learn words, My words, Add words, Words trainer are relative to active learn language which set in Settings.
        </CardText>
      </CardBody>
    </Card>
  </CardColumns>
</ContentComponent>
`;
